profile/Client/profileClient.go

```diff
@@ -285,36 +313,133 @@ type Frequency struct {
 	time string
 }
 
+//	func getTopMostFolder(path string) string {
+//		re := regexp.MustCompile(`^/([^/]+)`)
+//		match := re.FindStringSubmatch(path)
+//		if len(match) == 2 {
+//			return "/" + match[1]
+//		}
+//		return path
+//	}
+
+func isLaterTimestamp(timestamp1, timestamp2 string) bool {
+	t1, err := time.Parse(time.RFC3339, timestamp1)
+	if err != nil {
+		// Handle error, use some default value, or return false if you prefer
+		return false
+	}
+
+	t2, err := time.Parse(time.RFC3339, timestamp2)
+	if err != nil {
+		// Handle error, use some default value, or return false if you prefer
+		return false
+	}
+
+	return t1.After(t2)
+}
+
+// AggregateSummary used to aggregate summary data for a less cluttered view of file and process data
+func AggregateSummary(inputMap map[Profile]*Frequency, Operation string) map[Profile]*Frequency {
+	outputMap := make(map[Profile]*Frequency)
+	var fileArr []string
+	fileSumMap := make(map[Profile]*Frequency)
+	updatedSumMap := make(map[Profile]*Frequency)
+	if Operation == "Network" || Operation == "Syscall" {
+		return inputMap
+	}
+	for prof, count := range inputMap {
+		if Operation == "File" || Operation == "Process" {
+			fileArr = append(fileArr, prof.Resource)
+			fileSumMap[prof] = count
+		} else {
+			updatedSumMap[prof] = count
+		}
+	}
+	inputMap = updatedSumMap
+	aggregatedPaths := common.AggregatePaths(fileArr)
+	for summary, countTime := range fileSumMap {
+		for _, path := range aggregatedPaths {
+			if strings.HasPrefix(summary.Resource, path.Path) && (len(summary.Resource) == len(path.Path) || summary.Resource[len(strings.TrimSuffix(path.Path, "/"))] == '/') {
+				summary.Resource = path.Path
+				break
+			}
+		}
+		if existingFreq, ok := outputMap[summary]; ok {
+			// If the prof already exists, update the frequency and timestamp if needed
+			existingFreq.freq += countTime.freq
+
+			if isLaterTimestamp(countTime.time, existingFreq.time) {
+				existingFreq.time = countTime.time
+			}
+			outputMap[summary] = existingFreq
+		} else {
+			outputMap[summary] = countTime
+		}
+	}
+
+	return outputMap
+}
+
 func generateRowsFromData(data []pb.Log, Operation string) []table.Row {
 	var s SomeData
 	m := make(map[Profile]int)
 	w := make(map[Profile]*Frequency)
 	for _, entry := range data {
-
-		if (entry.Operation == Operation && entry.NamespaceName == o1.Namespace) ||
-			(entry.Operation == Operation && entry.PodName == o1.Pod) ||
-			(entry.Operation == Operation && len(o1.Namespace) == 0 && len(o1.Pod) == 0) {
-
-			p := Profile{
-				Namespace: entry.NamespaceName,
-				PodName:   entry.PodName,
-				Process:   entry.ProcessName,
-				Resource:  entry.Resource,
-				Result:    entry.Result,
+		if Operation == "File" || Operation == "Process" || Operation == "Network" {
```

Comment:
This whole code block needs refactoring. Lot of duplication code below in the context of 'syscall' operation.